### PR TITLE
Update: Convert to es6 classes (#6)

### LINKED ISSUE #6: ES6 Conversion required

### PR DESCRIPTION
**Pull request by joe-allen-89**
[//]: # (Please title your PR according to eslint commit conventions)
[//]: # (See https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-eslint#eslint-convention for details)

[//]: # (Link the PR to the original issue)
#6 

[//]: # (Delete Fix, Update, New and/or Breaking sections as appropriate)
### Update
* Convert to es6 classes (fixes #6)

### CODE REVIEW

File: js/adapt-drop.js [hunk 1]
```diff
@@ -1,93 +1,15 @@
-define([ 'core/js/adapt' ], function(Adapt) {
+import DropView from './DropView';
+import Adapt from 'core/js/adapt';
 
-  class DropView extends Backbone.View {
+Adapt.once('app:dataReady', () => {
```
Comment (oliverfoster) [inline js/adapt-drop.js]:
```js
class AdaptDrop extends Backbone.Controller {

  initialize() {
    this.listenTo(Adapt, 'app:dataReady', this.onDataReady);
  }
  
  onDataReady() {
    const views = [ 'menu', 'menuItem', 'page', 'article', 'block', 'component' ];
    const eventList = views.map(view => `${view}View:postRender`).join(' ')
    this.listenTo(Adapt, eventList, this.onEvent);
  }
  
  onEvent(model, $el) {
    const config = model.get('_drop');
    if (!config?._isEnabled) return;
    new DropView({ model, $content: $el.children().first() });
  }

}

export default new AdaptDrop();
```

File: js/DropView [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    this.toggle();
+
+    return false;
+  }
+
+  setChildrenVisibility(isVisible) {
+    this.model.getChildren().each(child => {
+      if (child.setOnChildren) {
+        child.setOnChildren('_isVisible', isVisible, { pluginName: '_drop' });
+      }
```
Comment (oliverfoster) [inline js/DropView]:
```suggestion
      if (!child.setOnChildren) return;
      child.setOnChildren('_isVisible', isVisible, { pluginName: '_drop' });
```

File: js/DropView [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    const modelClasses = this.model.get('_drop')._classes;
+
+    if (modelClasses) classes += ` ${modelClasses}`;
+
+    return classes;
+  }
+
+  initialize({ $content }) {
+    this.$content = $content;
+    this.listenTo(Adapt, 'remove', this.remove).render();
```
Comment (oliverfoster) [inline js/DropView]:
```suggestion
    this.listenTo(Adapt, 'remove', this.remove);
    this.render();
```

or

```js
    this
      .listenTo(Adapt, 'remove', this.remove)
      .render();
```

File: js/DropView [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  events() {
+    return { 'click .js-toggle-view': 'onClick' };
+  }
+
+  render() {
+    const template = Handlebars.templates.drop;
+    const data = this.model.get('_drop');
+
+    this.$el.html(template(data)).appendTo(this.$content.parent());
+    this.$content.addClass('drop__content').appendTo(this.$el).hide();
```
Comment (oliverfoster) [inline js/DropView]:
```suggestion
    this.$el
      .html(template(data))
      .appendTo(this.$content.parent());
    this.$content
      .addClass('drop__content')
      .appendTo(this.$el)
      .hide();
```

File: js/DropView [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  toggle() {
+    const shouldOpen = !this.$el.hasClass('is-open');
+
+    this.$el.toggleClass('is-open is-closed');
+
+    this.$('.drop__btn').first()
+      .addClass('is-visited')
+      .attr('aria-expanded', shouldOpen);
+
+    this.$content.stop().slideToggle({
```
Comment (oliverfoster) [inline js/DropView]:
```suggestion
    this.$content
      .stop()
      .slideToggle({
```

File: js/DropView (hunk 3 above)
Comment (joe-allen-89) [reply js/DropView]: Second option looks a bit cleaner to me.

File: js/adapt-drop.js (hunk 1 above)
Comment (joe-allen-89) [reply js/adapt-drop.js]: @oliverfoster implemented with one difference, curly braces needed in onEvent function to access model & $el
`onEvent({model, $el})`

**Review by oliverfoster — APPROVED** [review]

**Pull request merged**